envoy/http/query_params.h

```diff
@@ -15,6 +18,22 @@ namespace Utility {
 using QueryParams = std::map<std::string, std::string>;
 using QueryParamsVector = std::vector<std::pair<std::string, std::string>>;
 
+class QueryParamsMulti {
+private:
+  std::map<std::string, std::vector<std::string>> data;
```

Comment:
I am wondering why the sorted map is used here? Considering unordered map if sorting is not needed.

Also,  please use absl map:
unordered : `absl::flat_hash_map` OR `absl::node_hash_map` if you need pointer stability for key (which not seems to be the case here)
ordered: absl::btree_map